File: fixtures/sample.java
// --- FMM ---
// exports: [DataProcessor, ProcessConfig, Repository, Status, process, transform]
// imports: [java.util, org.springframework]
// dependencies: [java.util.List, java.util.Map, java.util.Optional, org.springframework.stereotype.Service]
// annotations: [Deprecated, FunctionalInterface, Override, Service]
// ---

package com.example.service;

import java.util.List;
import java.util.Map;
import java.util.Optional;
import org.springframework.stereotype.Service;

@Service
public class DataProcessor {
    private final Map<String, Object> cache;

    public DataProcessor() {
        this.cache = Map.of();
    }

    public List<String> process(List<String> input) {
        return input.stream()
            .filter(s -> !s.isEmpty())
            .toList();
    }

    public Optional<String> transform(String value) {
        if (value == null || value.isBlank()) {
            return Optional.empty();
        }
        return Optional.of(value.toUpperCase());
    }

    private void validate(String input) {
        if (input == null) {
            throw new IllegalArgumentException("Input cannot be null");
        }
    }

    @Override
    public String toString() {
        return "DataProcessor{}";
    }
}

public interface Repository<T> {
    T findById(long id);
    List<T> findAll();
    void save(T entity);
}

@Deprecated
public enum Status {
    ACTIVE,
    INACTIVE,
    PENDING
}

@FunctionalInterface
public interface ProcessConfig {
    void configure(Map<String, Object> settings);
}

File: fixtures/sample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MyApp.Services
{
    [Serializable]
    public class DataService
    {
        private readonly Dictionary<string, object> _cache;

        public DataService()
        {
            _cache = new Dictionary<string, object>();
        }

        [Required]
        public List<string> Transform(List<string> input)
        {
            return input.Where(s => !string.IsNullOrEmpty(s)).ToList();
        }

        private void Validate(string input)
        {
            if (string.IsNullOrEmpty(input))
                throw new ArgumentException("Input required");
        }
    }

    public interface IRepository<T>
    {
        T FindById(int id);
        IEnumerable<T> FindAll();
        Task SaveAsync(T entity);
    }

    [Obsolete]
    public enum Status
    {
        Active,
        Inactive,
        Pending
    }
}

namespace MyApp.Models
{
    public class ProcessConfig
    {
        public string Name { get; set; }
        public int MaxRetries { get; set; }
        public bool Debug { get; set; }
    }

    internal class InternalHelper
    {
        internal void DoWork() { }
    }
}
